pkg/chunk/azure/blob_storage_client.go

```diff
@@ -214,3 +243,41 @@ func (b *BlobStorage) DeleteObject(ctx context.Context, blobID string) error {
 func (b *BlobStorage) PathSeparator() string {
 	return b.delimiter
 }
+
+// Validate the config.
+func (c *BlobStorageConfig) Validate() error {
+	if c.Environment != azureGlobal && c.Environment != azureChinaCloud && c.Environment != azureGermanCloud && c.Environment == azureUSGovernment {
```

Comment:
You can use `supportedEnvironments` here:

```
if !util.StringsContain(supportedEnvironments, c.Environment) {
```

Reply:
done